.github/workflows/changelog-updater.yml

```diff
@@ -12,30 +12,49 @@ permissions:
 jobs:
   update-changelog:
     runs-on: ubuntu-latest
+    timeout-minutes: 10
     steps:
       - uses: actions/checkout@v6
         with:
           token: ${{ secrets.I18N_PAT }}
 
-      - uses: pnpm/action-setup@v4
-
-      - name: Setup Node
-        uses: actions/setup-node@v6
+      - uses: ./.github/actions/setup-pnpm
         with:
-          node-version-file: '.nvmrc'
+          install: 'false'  # Don't need dependencies for this script
 
+      # Process commits and generate changelog with proper error handling
       - name: Update changelog from commits
         id: changelog
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GITHUB_REPOSITORY: ${{ github.repository }}
           COMMIT_SHAS: ${{ join(github.event.commits.*.id, ' ') }}
         run: |
-          # Process all commits and automatically detect associated PRs using GitHub API
-          node build/update-changelog.js $COMMIT_SHAS > docs/CHANGELOG.md.new
-          mv docs/CHANGELOG.md.new docs/CHANGELOG.md
+          set -e
+
+          # Generate new changelog
+          if node build/update-changelog.js $COMMIT_SHAS > docs/CHANGELOG.md.new; then
+            # Only update if the script succeeded
+            mv docs/CHANGELOG.md.new docs/CHANGELOG.md
+            echo "updated=true" >> $GITHUB_OUTPUT
+          else
+            echo "::error::Failed to update changelog"
+            rm -f docs/CHANGELOG.md.new
+            echo "updated=false" >> $GITHUB_OUTPUT
+            exit 1
+          fi
+
+      - name: Check for actual changes
+        id: check-changes
+        run: |
+          if git diff --quiet docs/CHANGELOG.md; then
+            echo "changed=false" >> $GITHUB_OUTPUT
+          else
+            echo "changed=true" >> $GITHUB_OUTPUT
+          fi
 
       - name: Commit files
+        if: steps.check-changes.outputs.changed == 'true'
```

Comment:
The auto-commit-action includes this logic (checking for changed files) already. Why add more?